spec/dynamoid_advanced_where/node_tests/equality_spec.rb

```diff
@@ -32,11 +33,25 @@
       expect(klass.where{ native_bool == true }.all).to match_array([item1])
     end
 
-    it "matches true" do
+    it "matches false" do
       expect(klass.where{ native_bool == false }.all).to match_array([item2])
     end
   end
 
+  describe "default boolean equality" do
+    let!(:item1) { klass.create(default_bool: true) }
+    let!(:item2) { klass.create(default_bool: false) }
+    let!(:item3) { klass.create(default_bool: nil) }
+
+    it "matches true" do
+      expect(klass.where{ default_bool == true }.all).to match_array([item1])
+    end
```

Comment:
should it also have a case for matches NIL/DNE?

Reply:
There is a nil item which doesn't match true or false. Can you clarify the other case you think would be helpful?

Reply:
I expected that testing equality for `nil` would also be relevant, but doesn't look like that's the case.

Reply:
the test for nil is the existence spec spec/dynamoid_advanced_where/node_tests/exists_spec.rb

The tests are organized by operator not by field type, which may or may not be correct :/